apriltag/src/main/java/edu/wpi/first/apriltag/AprilTagFieldLayout.java

```diff
@@ -256,6 +261,57 @@ public int hashCode() {
     return Objects.hash(m_apriltags, m_origin);
   }
 
+  public static final class AProto
+      implements Protobuf<AprilTagFieldLayout, ProtobufAprilTagFieldLayout> {
+    @Override
+    public Class<AprilTagFieldLayout> getTypeClass() {
+      return AprilTagFieldLayout.class;
+    }
+
+    @Override
+    public Descriptor getDescriptor() {
+      return ProtobufAprilTagFieldLayout.getDescriptor();
+    }
+
+    @Override
+    public Protobuf<?, ?>[] getNested() {
+      return new Protobuf<?, ?>[] {AprilTag.proto, Pose3d.proto, FieldDimensions.proto};
+    }
+
+    @Override
+    public ProtobufAprilTagFieldLayout createMessage() {
+      return ProtobufAprilTagFieldLayout.newInstance();
+    }
+
+    @Override
+    public AprilTagFieldLayout unpack(ProtobufAprilTagFieldLayout msg) {
+      ArrayList<AprilTag> tags = new ArrayList<>(msg.getApriltags().length());
+      for (ProtobufAprilTag tag : msg.getApriltags()) {
+        tags.add(AprilTag.proto.unpack(tag));
+      }
+
+      var layout =
+          new AprilTagFieldLayout(tags, FieldDimensions.proto.unpack(msg.getFieldDimensions()));
+      layout.setOrigin(Pose3d.proto.unpack(msg.getOrigin()));
+
+      return layout;
+    }
+
+    @Override
+    public void pack(ProtobufAprilTagFieldLayout msg, AprilTagFieldLayout value) {
+      var tags = msg.getMutableApriltags().reserve(value.m_apriltags.size());
```

Comment:
Does the `AprilTags` parameter get cleared before this? (That is, will multiple calls to pack on the same message lead to only the most recent tags in the list?) I don't know enough about QuickBuf to know if this is the case.